### PR TITLE
[Concept Entry] Swift Subscripts

### LINKED ISSUE #2214: [Concept Entry] Swift Subscripts
### Reason/inspiration (optional)

We would like a new concept entry for Subscripts in Swift.

The entry should go in an file in docs/content/swift/concepts/subscripts/subscripts.md

The entry should: 

- Explain what subscripts are used for in Swift, starting with an introductory paragraph (2-3 sentences).
- Go over how to create a subscript with the keyword `subscript`.
- Links out to entries (like classes, structures, and enums) as needed.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by gracekishino**
Add an entry for Swift Subscripts

### Description

Added new folder `subscripts` inside `content/swiftui/concepts/`.
Added new entry file `content/swiftui/concepts/subscripts/subscripts.md` for SwiftUI Subscripts.

Closes #2214

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@gracekishino there's only one change I'd like to see. We generally prefer the output from examples illustrated in a separate ` ```shell ` block rather than a comment within the code. Like this:

````
This example results in the following output:

```shell
Output
```
````

Can you rewrite the examples to show the outputs like this?

**Comment by gracekishino** [comment]
@SSwiniarski thank you for the review! I moved the examples to separate shell blocks.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@gracekishino Looks good for a second review.

**Review by SSwiniarski — APPROVED** [review]

File: content/swift/concepts/subscripts/subscripts.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+---
+Title: 'Subscripts'
+Description: 'Subscripts are used to get and set values in collections such as arrays and dictionaries.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Subscripts'
```
Comment (caupolicandiaz) [inline content/swift/concepts/subscripts/subscripts.md:8]:
```suggestion
  - 'Collections'
```

Comment (caupolicandiaz) [reply content/swift/concepts/subscripts/subscripts.md:8]: subscripts isn't currently included in our tags list

File: content/swift/concepts/subscripts/subscripts.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+**Subscripts** are used to get and set values in collections such as [arrays](https://www.codecademy.com/resources/docs/swift/arrays) and [dictionaries](https://www.codecademy.com/resources/docs/swift/dictionaries). It's also possible to define one or more subscripts for a [class](https://www.codecademy.com/resources/docs/swift/classes), [structure](https://www.codecademy.com/resources/docs/swift/structures), or [enum](https://www.codecademy.com/resources/docs/swift/enums) so values can easily be retrieved or set using an index or key, rather than using methods.
+
+## Using Subscripts to Get and Set Values
+
+### Syntax
```
Comment (caupolicandiaz) [inline content/swift/concepts/subscripts/subscripts.md:19]:
```suggestion
```

File: content/swift/concepts/subscripts/subscripts.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+```
+
+New values are set using the same notation:
+
+```pseudo
+array[index] = new value
+dictionary[key] = new value
+```
+
+### Examples
```
Comment (caupolicandiaz) [inline content/swift/concepts/subscripts/subscripts.md:35]:
```suggestion
### Getting and Setting Examples
```

File: content/swift/concepts/subscripts/subscripts.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+var gardenInfo = ["name": "Happy Herb Garden", "type": "Herb", "length": "1m"]
+
+// Set the element with the key "length" to "2m", replacing "1m".
+gardenInfo["length"] = "2m"
+
+// Print the element with the key "length".
+print(gardenInfo["length"]!)
+```
+
+> **Note:** The "!" forces the optional String? value to unwrap
```
Comment (caupolicandiaz) [inline content/swift/concepts/subscripts/subscripts.md:69]:
```suggestion
> **Note:** The "!" forces the optional String? value to unwrap.
```

File: content/swift/concepts/subscripts/subscripts.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+This example will output the following:
+
+```shell
+2m
+```
+
+## Creating Subscripts
+
+### Syntax
```
Comment (caupolicandiaz) [inline content/swift/concepts/subscripts/subscripts.md:79]:
```suggestion
```

File: content/swift/concepts/subscripts/subscripts.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+If a subscript is read-only, the code for retrieving the value can be entered directly inside the `subscript {}` wrapper, omitting the `get {}` wrapper.
+
+```pseudo
+subscript(input parameters) -> return type {
+        return the value here
+}
+```
+
+### Example
```
Comment (caupolicandiaz) [inline content/swift/concepts/subscripts/subscripts.md:104]:
```suggestion
### Creating Subscripts Example
```

File: content/swift/concepts/subscripts/subscripts.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+The example below defines a read-only subscript within a [`struct`](https://www.codecademy.com/resources/docs/swift/structures) called `FlowerGarden`.
```
Comment (caupolicandiaz) [inline content/swift/concepts/subscripts/subscripts.md:106]: this topic is already linked above, this link seems superfluous

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@gracekishino I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

File: content/swift/concepts/subscripts/subscripts.md (hunk 7 above)
Comment (gracekishino) [reply content/swift/concepts/subscripts/subscripts.md]: I removed this superfluous link.

**Comment by gracekishino** [comment]
@caupolicandiaz thanks so much for reviewing! I updated the changes.

**Review by caupolicandiaz — APPROVED** [review]
@gracekishino @SSwiniarski Looks good.

**Comment by SSwiniarski** [comment]
@gracekishino Congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/swift/subscripts

image

**Comment by gracekishino** [comment]
@SSwiniarski @caupolicandiaz thanks so much for your support with this entry!

**Pull request merged**